swupd/create_manifests_test.go

```diff
@@ -49,81 +48,46 @@ func TestInitBuildDirs(t *testing.T) {
 }
 
 func TestCreateManifests(t *testing.T) {
-	var err error
 	testDir := mustSetupTestDir(t, "basic")
 	defer removeIfNoErrors(t, testDir)
 	mustInitStandardTest(t, testDir, "0", "10", []string{"test-bundle"})
 	mustGenFile(t, testDir, "10", "test-bundle", "foo", "content")
 	mustCreateManifestsStandard(t, 10, testDir)
 
-	// set last version to 10
-	mustInitStandardTest(t, testDir, "10", "20", []string{"test-bundle"})
-	mustGenFile(t, testDir, "20", "test-bundle", "foo", "new content")
-	mustCreateManifestsStandard(t, 20, testDir)
-
-	lines := []struct {
-		sub      []byte
-		expected bool
-	}{
-		{[]byte("MANIFEST\t1"), true},
-		{[]byte("version:\t10"), true},
-		{[]byte("previous:\t0"), true},
-		{[]byte("filecount:\t5"), true},
-		{[]byte("timestamp:\t"), true},
-		{[]byte("contentsize:\t"), true},
-		{[]byte("includes:\tos-core"), true},
-		{[]byte("10\t/foo"), true},
-		{[]byte("\t0\t/foo"), false},
-		{[]byte("10\t/usr/share"), true},
-		{[]byte(".d..\t"), false},
+	expSubs := []string{
+		"MANIFEST\t1",
+		"version:\t10",
+		"previous:\t0",
+		"filecount:\t5",
+		"timestamp:\t",
+		"contentsize:\t",
+		"includes:\tos-core",
+		"10\t/foo",
+		"10\t/usr/share",
 	}
+	checkManifestContains(t, testDir, "10", "test-bundle", expSubs, true)
 
-	// do not use the fileContains helper for this because we are checking a lot
-	// of strings in the same file
-	b, err := ioutil.ReadFile(filepath.Join(testDir, "www/10/Manifest.test-bundle"))
-	if err != nil {
-		t.Fatal("could not read test file for checking")
+	nExpSubs := []string{
+		"\t0\t/foo",
+		".d..\t",
 	}
+	checkManifestContains(t, testDir, "10", "test-bundle", nExpSubs, false)
```

Comment:
It will be better for our sanity if we had a pair of functions: checkManifestContains and checkManifestNotContains (not sure this is proper english). I know what you mean here with the false in the end of the call, but will forget soon.

(Other solutions that make explicit Contains/NotContains distinction are also valid)